src/common/input/config_schema.json

```diff
@@ -0,0 +1,75 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "zellij/config_schema.json",
+    "description": "Configuration for the zellij terminal multiplexer",
+    "type": "object",
+    "properties": {
+        "keybinds": {
+            "description": "user-defined keybinds",
+            "type": "object",
+            "properties": {
+                "unbind": {
+                    "$ref": "#/$defs/unbind"
+                },
+                "normal": {
+                    "$ref": "#/$defs/mode_keybinds"
+                },
+                "locked": {
+                    "$ref": "#/$defs/mode_keybinds"
+                },
+                "resize": {
+                    "$ref": "#/$defs/mode_keybinds"
+                },
+                "panes": {
+                    "$ref": "#/$defs/mode_keybinds"
+                },
+                "tab": {
+                    "$ref": "#/$defs/mode_keybinds"
+                },
+                "scroll": {
+                    "$ref": "#/$defs/mode_keybinds"
+                },
+                "renametab": {
+                    "$ref": "#/$defs/mode_keybinds"
+                }
+            }
+        }
+    },
+    "$defs": {
+        "mode_keybinds": {
+            "description": "the set of keybinds for a particular input mode",
+            "type": "array",
+            "items": {
+                "anyOf": [
+                    {
+                        "description": "an individual keybind",
+                        "properties": {
+                            "action": {
+                                "description": "the actions to be executed, in order",
+                                "type": "array"
+                            },
+                            "key": {
+                                "description": "the key sequence(s) which invoke this keybind",
+                                "type": "array"
+                            }
+                        },
+                        "required": [
+                            "action",
+                            "key"
+                        ]
+                    },
+                    {
+                        "properties": {
+                            "unbind": {
+                                "$ref": "#/$defs/unbind"
+                            }
+                        }
+                    }
+                ]
+            }
+        },
+        "unbind": {
+            "description": "unbind a particular key"
```

Comment:
I definitely want these kind of things in here, but haven't yet been successful in applying them. I know that YAML has some peculiar habits with types (see [my colleague's blogpost](https://www.lewisgaul.co.uk/blog/coding/2021/04/18/zig-nestedtext/)) but I'm also just not very familiar with it.

Feel free to play around with it yourself, if you get time. Otherwise I'll pick it up when I'm doing zellij stuff and finished with anything Kunal needs.